### DISCUSSION TITLE
[Duda T0-Parte-2] Give-puffles error en taste case medium/hard-2

### DISCUSSION THREAD

**Discussion by Anehg** [IIC2133-PUC/2024-2#36] [Tarea 0] [answered]
Hola,

Me surgió un error al usar los test cases de nivel medio y avanzado de la parte 2. El problema es que al ejecutar el código ocurre un "segmentation fault" debido a que el rango especificado para el comando GIVE-PUFFLES está fuera del rango de la lista de puffles del pingüino donante. Mi pregunta es si se debe comprobar que el rango dado esté dentro del rango de puffles del donante, ya que en el enunciado no se menciona esta verificación y, en estos casos, el receptor podría recibir los puffles.

Como ejemplo, el error se produce en el archivo Part-2-medium-2/input.txt en la línea 1560, con el comando GIVE-PUFFLES 20 30 29 41, dado que el pingüino 20 no tiene índices del 29 al 41 (de hecho, según lo que tengo, solo tiene 24 puffles). El pingüino 30 tiene 37 puffles, por lo que podría recibir 12 puffles más, alcanzando un total de 49 puffles, dentro del límite permitido de 90 puffles.

Al final, también podría ser un error en mi código (aunque al ejecutarlo con Valgrind no tengo memory leaks, así que no es por eso). Quedo atento a la respuesta.

**Comment by JavierLarre** [comment, marked as answer]
Hola! Los rangos dados son dentro del rango del inventario del pingüino, por lo tanto tu problema debe ser causado por alguna otra parte de la lógica de tu código. Los otros comandos como steal que modifican el inventario pueden ser la causa, te recomiendo aislar y recrear el problema.
